tests/engines/collections/acceptance/submit/submit-test.ts

```diff
@@ -30,33 +33,164 @@ module('Collections | Acceptance | submit', hooks => {
             primaryCollection,
             licensesAcceptable,
         });
+
+        const store = this.owner.lookup('service:store');
+
+        const newTitle = 'New Title';
+        const newDescription = 'New description.';
+
         await visit(`/collections/${provider.id}/submit`);
+
+        /* Select a project */
+
         await percySnapshot('Collections | Acceptance | submit | select project');
+
+        // open item picker
         await untrackedClick('[data-test-collections-item-picker] .ember-power-select-trigger');
-        await untrackedClick(`[data-test-collections-node-title="${nodeToBeAdded.title}"]`);
+        // select node
+        const nodeToBeAddedOption = document.querySelector(
+            `[data-test-collections-node-title="${nodeToBeAdded.title}"]`,
+        );
+        if (nodeToBeAddedOption) {
+            await untrackedClick(nodeToBeAddedOption);
+        }
+
+        /* Project metadata */
 
         await percySnapshot('Collections | Acceptance | submit | project metadata');
-        await untrackedClick('[data-test-collections-license-picker] .ember-power-select-trigger');
-        await untrackedClick('.ember-power-select-option');
-        await untrackedClick('[data-test-project-metadata-continue]');
+
+        await fillIn('[data-test-project-metadata-title] input', newTitle);
+        await fillIn('[data-test-project-metadata-description] textarea', newDescription);
+        // open license picker
+        await untrackedClick('[data-test-project-metadata-license-picker] .ember-power-select-trigger');
+        // select first license
+        const firstLicenseOption = document.querySelector('.ember-power-select-option');
+        if (firstLicenseOption) {
+            await untrackedClick(firstLicenseOption);
+        }
+        // remove third tag
+        await untrackedClick(`[data-test-project-metadata-tag="${nodeToBeAdded.tags[2]}"] + .emberTagInput-remove`);
+        await untrackedClick('[data-test-project-metadata-save-button]');
+
+        assert.dom('[data-test-project-metadata-complete-title-value]')
+            .hasText(newTitle, 'title is updated');
+        assert.dom('[data-test-project-metadata-complete-description-value]')
+            .hasText(newDescription, 'description is updated');
+        const collectionProvider: CollectionProvider = store.peekRecord('collection-provider', provider.id);
+        const licenses = await collectionProvider.get('licensesAcceptable');
+        const firstLicense = licenses.get('firstObject');
+        assert.dom('[data-test-project-metadata-complete-license-name-value]')
+            .hasText(firstLicense ? firstLicense.name : '', 'license is updated');
```

Comment:
Is there a simple way to assure that `firstLicense` has property `name`? If not, I think this assertion can stay as it is.

Reply:
This is testing that `firstLicense` is defined because the type of `licenses.get('firstObject')` is `License | undefined`.